src/main/java/org/stellar/sdk/InvokeHostFunctionOperation.java

```diff
@@ -0,0 +1,151 @@
+    InvokeHostFunctionOp op = new InvokeHostFunctionOp();
+    op.setHostFunction(this.hostFunction);
+    op.setAuth(this.auth);
+
+    org.stellar.sdk.xdr.Operation.OperationBody body =
+        new org.stellar.sdk.xdr.Operation.OperationBody();
+    body.setDiscriminant(OperationType.INVOKE_HOST_FUNCTION);
+    body.setInvokeHostFunctionOp(op);
+
+    return body;
+  }
+
+  /**
+   * Builds InvokeHostFunction operation.
+   *
+   * @see InvokeHostFunctionOperation
+   */
+  public static class Builder {
+    private final HostFunction hostFunction;
+    private final SorobanAuthorizationEntry[] auth;
+
+    private String mSourceAccount;
+
+    /**
+     * Creates a new InvokeHostFunction builder using the specified parameters.
+     *
+     * @param op the XDR representation of the {@link InvokeHostFunctionOperation}.
+     */
+    Builder(InvokeHostFunctionOp op) {
+      this.hostFunction = op.getHostFunction();
+      this.auth = op.getAuth();
+    }
+
+    /**
+     * Creates a new InvokeHostFunction builder using the specified parameters.
+     *
+     * @param hostFunction host function to execute (with its wrapped parameters)
+     * @param auth the authorizations required to execute the host function
+     */
+    public Builder(HostFunction hostFunction, SorobanAuthorizationEntry[] auth) {
+      this.hostFunction = checkNotNull(hostFunction, "hostFunction cannot be null");
+      this.auth = checkNotNull(auth, "auth cannot be null");
+    }
+
+    /**
+     * Creates a new InvokeHostFunction builder using the specified parameters.
+     *
+     * @param hostFunction host function to execute (with its wrapped parameters)
+     */
+    public Builder(HostFunction hostFunction) {
+      this.hostFunction = checkNotNull(hostFunction, "hostFunction cannot be null");
+      this.auth = new SorobanAuthorizationEntry[] {};
+    }
+
+    /**
+     * Set source account of this operation
+     *
+     * @param sourceAccount Source account
+     * @return Builder object so you can chain methods.
+     */
+    public Builder setSourceAccount(String sourceAccount) {
+      mSourceAccount = checkNotNull(sourceAccount, "sourceAccount cannot be null");
+      return this;
+    }
+
+    /** Builds an operation */
+    public InvokeHostFunctionOperation build() {
+      InvokeHostFunctionOperation operation =
+          new InvokeHostFunctionOperation(this.hostFunction, this.auth);
+      if (mSourceAccount != null) {
+        operation.setSourceAccount(mSourceAccount);
+      }
+      return operation;
+    }
+  }
+
+  public int hashCode() {
+    return Objects.hashCode(hostFunction, auth);
+  }
+
+  @Override
+  public boolean equals(Object object) {
+    if (!(object instanceof InvokeHostFunctionOperation)) {
+      return false;
+    }
+
+    InvokeHostFunctionOperation o = (InvokeHostFunctionOperation) object;
+    return Objects.equal(this.getHostFunction(), o.getHostFunction())
```

Comment:
I noticed that in all the operations, there was no comparison made with the source. However, the source is a part of the Op. I believe we should address this issue in another PR.